tests/modules/test_crop.py

```diff
@@ -282,6 +282,19 @@ def test_04_06_crop_color_with_rectangle(self):
         output_image = workspace.image_set.get_image(OUTPUT_IMAGE)
         self.assertTrue(np.all(output_image.pixel_data == expected_image))
 
+    def test_04_07_crop_with_rectangle_float_bounds(self):
+        x, y = np.mgrid[0:10, 0:10]
+        input_image = (x / 100.0 + y / 10.0).astype(np.float32)
+        expected_image = input_image[2:8, 1:9]
+        workspace, module = self.make_workspace(input_image)
+        module.shape.value = cpmc.SH_RECTANGLE
+        module.horizontal_limits.set_value((1.2, 9.0000003))
+        module.vertical_limits.set_value((2.5, 8.999999))
```

Comment:
I'd expect this to round to 9, rather than 8. 

What about using `int(numpy.round(r[x]))` to round up or down?

Reply:
I was thinking that as well, but I was thinking more of the edge case:
If the left bound is 0.00001, it should go to index 0. If the right bound is 255.99999, and it gets rounded up to 256, that index would be out of bounds for an array of length 256. 

That being said, we're already ensuring the selection is within the bounds so I can change this to be more rational :+1:

Reply:
@mcquin Unfortunately, it looks like `vertical_limits` is of type `IntegerOrUnboundedRange`, so setting it via the `set_value` method automatically rounds the float down...Looks like this test didn't even sufficiently test my change! Do you know of another way we can set these values so that they can actually exist as floats at the time that `apply_rectangle_cropping` is called?

Reply:
Interesting! I suppose the solution would be to implement an equivalent `FloatOrUnboundedRange` setting. However, since the old behavior is to round down then your solution here is sufficient.